packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -296,43 +304,52 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 			// https://dev.azure.com/fluidframework/internal/_workitems/edit/779
 			await this.lockedSummaryAction(
 				() => {},
-				async () =>
-					this.refreshLatestSummaryAckCallback({
+				async () => {
+					const options: IRefreshSummaryAckOptions = {
 						proposalHandle: summaryOpHandle,
 						ackHandle: summaryAckHandle,
 						summaryRefSeq: refSequenceNumber,
 						summaryLogger,
-					}).catch(async (error) => {
-						// If the error is 404, so maybe the fetched version no longer exists on server. We just
-						// ignore this error in that case, as that means we will have another summaryAck for the
-						// latest version with which we will refresh the state. However in case of single commit
-						// summary, we might me missing a summary ack, so in that case we are still fine as the
-						// code in `submitSummary` function in container runtime, will refresh the latest state
-						// by calling `prefetchLatestSummaryThenClose`. We will load the next summarizer from the
-						// updated state and be fine.
-						const isIgnoredError =
-							isFluidError(error) &&
-							error.errorType === DriverErrorTypes.fileNotFoundOrAccessDeniedError;
-
-						summaryLogger.sendTelemetryEvent(
-							{
-								eventName: isIgnoredError
-									? "HandleSummaryAckErrorIgnored"
-									: "HandleLastSummaryAckError",
-								referenceSequenceNumber: refSequenceNumber,
-								proposalHandle: summaryOpHandle,
-								ackHandle: summaryAckHandle,
-							},
-							error,
-						);
-					}),
+					};
+					await this.refreshLatestSummaryAckCallback(options).catch(async (error) =>
```

Comment:
Why change this to an await?

Reply:
It was originally awaited, but I had to put it in a callstack block to centralize some code, so I put an await. Otherwise, we'll run into some race condition issues and the tests won't pass.